scene/main/node.cpp

```diff
@@ -1715,8 +1715,41 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = data.owner && (data.owner == p_parent || data.owner->is_ancestor_of(p_parent));
+	Node *owner_temp = data.owner;
+	List<const Node *> common_parents;
+
+	// If the new parent is related to the owner, find all children of the reparented node who have the same owner so that we can reassign them.
+	if (preserve_owner) {
+		List<const Node *> to_visit;
+
+		to_visit.push_back(this);
+		common_parents.push_back(this);
+
+		while (to_visit.size() > 0) {
+			const Node *check = to_visit.back()->get();
+			to_visit.pop_back();
+
+			for (int i = 0; i < check->get_child_count(); i++) {
+				const Node *child = check->get_child(i, false);
+				to_visit.push_back(child);
+				if (child->data.owner == owner_temp) {
+					common_parents.push_back(child);
+				}
+			}
+		}
+	}
+
 	data.parent->remove_child(this);
 	p_parent->add_child(this);
+
+	// Reassign the old owner to those found nodes.
+	if (preserve_owner) {
+		for (const Node *E : common_parents) {
+			Node *N = const_cast<Node *>(E);
+			N->set_owner(owner_temp);
+		}
```

Comment:
```suggestion
		for (Node *E : common_parents) {
			E->set_owner(owner_temp);
		}
```

Reply:
also done! just commenting on this for clarity's sake after the fact.